### PR TITLE
make test-e2e-node: default to containerd

### LINKED ISSUE #107856: make test_e2e_node fails by default with the docker deprecation.
### What happened?

$ make test-e2e-node

fails with an error `[7] E0129 22:41:05.743374  624494 test_context.go:496] Unsupported CRI container runtime: "docker"`

because of the docker deprecation. I think we have to fix the default options for helper scripts.

### What did you expect to happen?

$ make test-e2e-node

succeeds in common cases by default.

### How can we reproduce it (as minimally and precisely as possible)?

$ make test-e2e-node

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

/assign

### PR DESCRIPTION
**Pull request by endocrimes**
#### What type of PR is this?

/kind cleanup
/sig node
/cc @adisky @SergeyKanzhelev @dims 

#### What this PR does / why we need it:

Since removing dockershim, `make test-e2e-node` will fail by default as
there is no provided container runtime endpoint.

This commit defaults us to using containerd's default socket path as the
local test target, rather than failing hard.

#### Which issue(s) this PR fixes:

Fixes #107856

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: build/root/Makefile [hunk 1]
```diff
@@ -226,10 +226,9 @@ define TEST_E2E_NODE_HELP_INFO
 #  TIMEOUT: For REMOTE=true only. How long (in golang duration format) to wait
 #    for ginkgo tests to complete. Defaults to 45m.
 #  PARALLELISM: The number of ginkgo nodes to run.  Defaults to 8.
-#  RUNTIME: Container runtime to use (eg. docker, remote).
-#    Defaults to "docker".
 #  CONTAINER_RUNTIME_ENDPOINT: remote container endpoint to connect to.
 #    Used when RUNTIME is set to "remote".
```
Comment (gjkim42) [inline build/root/Makefile:230]:
```suggestion
```

File: build/root/Makefile [hunk 2]
```diff
@@ -226,10 +226,9 @@ define TEST_E2E_NODE_HELP_INFO
...
+#    Defaults to "/run/containerd/containerd.sock".
 #  IMAGE_SERVICE_ENDPOINT: remote image endpoint to connect to, to prepull images.
 #    Used when RUNTIME is set to "remote".
```
Comment (gjkim42) [inline build/root/Makefile:233]:
```suggestion
```

**Review by gjkim42 — COMMENTED** [review]
nit: I guess there is no place that uses "RUNTIME" environment variable.

**Review by adisky — COMMENTED** [review]
/lgtm
(with @gjkim42 comments addressed)
We already removed `RUNTIME` in https://github.com/kubernetes/kubernetes/pull/107663 to fix #107856 but it makes sense to add containerd as default

**Comment by endocrimes** [comment]
Updated 👍

**Comment by ehashman** [comment]
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv1

(unrelated, just testing a job change)

File: build/root/Makefile (hunk 2 above)
Comment (ehashman) [reply build/root/Makefile:232]: Agree with this suggestion

**Review by ehashman — COMMENTED** [review]
/priority important-soon
/triage accepted
/lgtm

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

thank you!

**Pull request merged**